File: app/src/main/java/com/movieapplication/model/tvseries/TvSerieRepository.java
package com.movieapplication.model.tvseries;

import android.app.Application;

import androidx.lifecycle.MutableLiveData;

import com.movieapplication.R;
import com.movieapplication.model.tvseries.Result;
import com.movieapplication.service.MovieDataService;
import com.movieapplication.service.RetrofitInstance;
import com.movieapplication.service.TvSeriesDataService;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class TvSerieRepository {
    private ArrayList<Tvserie> tvseries = new ArrayList<>();
    private MutableLiveData<List<Tvserie>> mutableLiveData = new MutableLiveData<>();
    private Application application;

    public TvSerieRepository(Application application) {
        this.application = application;
    }


    public MutableLiveData<List<Tvserie>> getMutableLiveData(String listType) {
        TvSeriesDataService tvSeriesDataService = RetrofitInstance.getServiceTvSerie();

        Call<Result> call = tvSeriesDataService.getTopRatedTvSeries(application.getApplicationContext().getString(R.string.api_key));

        call.enqueue(new Callback<Result>() {
            @Override
            public void onResponse(Call<Result> call, Response<Result> response) {
                Result result = response.body();

                if (result != null && result.getResults() != null) {
                    tvseries = (ArrayList<Tvserie>) result.getResults();
                    mutableLiveData.setValue(tvseries);
                }
            }

            @Override
            public void onFailure(Call<Result> call, Throwable t) {

            }
        });

        return mutableLiveData;
    }

}

File: app/src/main/java/com/movieapplication/model/tvseries/TvSerieRepository.kt
package com.movieapplication.model.tvseries

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.movieapplication.R
import com.movieapplication.service.RetrofitInstance
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response


class TvSerieRepository(private val application: Application) {
    private var tvseries = ArrayList<Tvserie>()
    private val mutableLiveData = MutableLiveData<List<Tvserie>>()
    fun getMutableLiveData(listType: String?): MutableLiveData<List<Tvserie>> {
        val tvSeriesDataService = RetrofitInstance.getServiceTvSerie()
        val call = tvSeriesDataService.getTopRatedTvSeries(
            application.applicationContext.getString(R.string.api_key)
        )
        call!!.enqueue(object : Callback<Result?> {
            override fun onResponse(call: Call<Result?>, response: Response<Result?>) {
                val result = response.body()
                if (result != null && result.results != null) {
                    tvseries = result.results as ArrayList<Tvserie>
                    mutableLiveData.value = tvseries
                }
            }

            override fun onFailure(call: Call<Result?>, t: Throwable) {}
        })
        return mutableLiveData
    }
}